diff --git a/src/TritonRoute/src/pa/FlexPA_prep.cpp b/src/TritonRoute/src/pa/FlexPA_prep.cpp
index da3790cb1..4217e9216 100644
--- a/src/TritonRoute/src/pa/FlexPA_prep.cpp
+++ b/src/TritonRoute/src/pa/FlexPA_prep.cpp
@@ -981,6 +981,11 @@ void FlexPA::prepPoint_pin_checkPoints(
   for (int i = 0; i < (int) layerPolysets.size(); i++) {
     layerPolysets[i].get_polygons(layerPolys[i]);
   }
+  if (aps.size() > 150) {
+    cout << "too many aps (" << aps.size() << ")" << endl;
+    cout << "stripping " << instTerm->getInst()->getName() << endl;
+    aps.resize(150);
+  }
   for (auto& ap : aps) {
     auto layerNum = ap->getLayerNum();
     frPoint pt;
